chaincode/marbles_transfer/marbles_transfer.go

```diff
@@ -361,9 +395,44 @@ func transferMarbleState(ctx contractapi.TransactionContextInterface, marble *Ma
 		return fmt.Errorf("failed to put Marble private properties for buyer: %s", err.Error())
 	}
 
-	// TODO delete the price records for buyer and seller
+	// Delete the price records for seller
+	marblePriceKey, err := ctx.GetStub().CreateCompositeKey(typeMarbleForSale, []string{marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key: %s", err.Error())
+	}
 
-	// TODO add a state record for a 'receipt' in both buyer and seller private data collection to record the sales price and date
+	err = ctx.GetStub().DelPrivateData(collectionSeller, marblePriceKey)
+	if err != nil {
+		return fmt.Errorf("failed to delete marble price from implicit private data collection for seller: %s", err.Error())
+	}
+
+	// Delete the price records for buyer
+	marblePriceKey, err = ctx.GetStub().CreateCompositeKey(typeMarbleBid, []string{marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key: %s", err.Error())
+	}
+
+	err = ctx.GetStub().DelPrivateData(collectionBuyer, marblePriceKey)
+	if err != nil {
+		return fmt.Errorf("failed to delete marble price from implicit private data collection for buyer: %s", err.Error())
+	}
+
+	// Keep record for a 'receipt' in both buyer and seller private data collection to record the sales price and date
+	// Persist the agreed to price in a collection sub-namespace based on receipt key prefix
+	receipt, err := ctx.GetStub().CreateCompositeKey(typeMarbleReceipt, []string{marble.ID})
+	if err != nil {
+		return fmt.Errorf("failed to create composite key: %s", err.Error())
+	}
+
+	err = ctx.GetStub().PutPrivateData(collectionBuyer, receipt, immutablePropertiesJSON)
```

Comment:
How about receiptKey instead of receipt?
And instead of saving the immutable properties JSON, I was thinking the sales price and sale date (GetTxTimestamp()) would get stored in a new receipt JSON.

Reply:
I like this. 🛠️